src/CosmosEditorManager.ts

```diff
@@ -21,153 +22,153 @@ import { MongoDocumentTreeItem } from './mongo/tree/MongoDocumentTreeItem';
-    convertToString(data: T): string;
+  label: string;
+  id: string;
+  getData(context: IActionContext): Promise<T>;
+  update(data: T, context: IActionContext): Promise<T>;
+  convertFromString(data: string): T;
+  convertToString(data: T): string;
 }
 
 export interface ShowEditorDocumentOptions {
-    /**
-     * Shows the document to the right of the current editor, and keeps focus on the active document
-     */
-    showInNextColumn?: boolean;
+  /**
+   * Shows the document to the right of the current editor, and keeps focus on the active document
+   */
+  showInNextColumn?: boolean;
 }
 
 export class CosmosEditorManager {
-    private fileMap: { [key: string]: ICosmosEditor } = {};
-    private ignoreSave: boolean = false;
-
-    private readonly showSavePromptKey: string = 'cosmosDB.showSavePrompt';
-    private _globalState: vscode.Memento;
-    private readonly _persistedEditorsKey: string = "ms-azuretools.vscode-cosmosdb.editors";
-
-    constructor(globalState: vscode.Memento) {
-        this._globalState = globalState;
+  private fileMap: { [key: string]: ICosmosEditor } = {};
+  private ignoreSave: boolean = false;
+
+  private readonly showSavePromptKey: string = 'cosmosDB.showSavePrompt';
+  private _globalState: vscode.Memento;
+  private readonly _persistedEditorsKey: string = "ms-azuretools.vscode-cosmosdb.editors";
+
+  constructor(globalState: vscode.Memento) {
+    this._globalState = globalState;
+  }
+
+  public async showDocument(context: IActionContext, editor: ICosmosEditor, fileName: string, options?: ShowEditorDocumentOptions): Promise<void> {
+    let column: vscode.ViewColumn = vscode.ViewColumn.Active;
+    let preserveFocus: boolean = false;
+    if (options && options.showInNextColumn) {
+      preserveFocus = true;
+      const activeEditor = vscode.window.activeTextEditor;
+      if (activeEditor && activeEditor.viewColumn >= vscode.ViewColumn.One) {
+        column = activeEditor.viewColumn < ViewColumn.Three ? activeEditor.viewColumn + 1 : ViewColumn.One;
+      }
     }
 
-    public async showDocument(context: IActionContext, editor: ICosmosEditor, fileName: string, options?: ShowEditorDocumentOptions): Promise<void> {
-        let column: vscode.ViewColumn = vscode.ViewColumn.Active;
-        let preserveFocus: boolean = false;
-        if (options && options.showInNextColumn) {
-            preserveFocus = true;
-            const activeEditor = vscode.window.activeTextEditor;
-            if (activeEditor && activeEditor.viewColumn >= vscode.ViewColumn.One) {
-                column = activeEditor.viewColumn < ViewColumn.Three ? activeEditor.viewColumn + 1 : ViewColumn.One;
-            }
-        }
-
-        const localDocPath = path.join(os.tmpdir(), 'vscode-cosmosdb-editor', fileName);
-        await fse.ensureFile(localDocPath);
+    const localDocPath = path.join(os.tmpdir(), 'vscode-cosmosdb-editor', filenamify(fileName, { maxLength: 255, replacement: "--" }));
```

Comment:
We try not to add third party dependencies if possible and I'm guessing we could fix the bug ourselves in a few lines of code without using `filenamify`

Reply:
I use Windows and am completely in the dark about what characters are invalid for a filename on all the platforms supported by the extension. Is there a function already in the extension for handling that? Is there an authoritative list available?

Reply:
Seems like mac/linux are much less restrictive than windows, so I wouldn't really worry about them. https://en.wikipedia.org/wiki/Filename

The file name here is purely for readability and doesn't affect the functionality, so we could also just remove any non-alphanumeric character to cover all bases